quarkus-extension/engine/qa/src/test/java/org/camunda/bpm/engine/cdi/test/QuarkusAutomaticDeployment.java

```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.cdi.test;
+
+import org.camunda.bpm.engine.cdi.test.api.BusinessProcessBeanTest;
+import org.camunda.bpm.engine.cdi.test.api.annotation.CompleteTaskTest;
+import org.camunda.bpm.engine.cdi.test.api.annotation.ProcessVariableLocalTypedTest;
+import org.camunda.bpm.engine.cdi.test.api.annotation.ProcessVariableTypedTest;
+import org.camunda.bpm.engine.cdi.test.bean.QuarkusDeclarativeProcessController;
+import org.camunda.bpm.engine.cdi.test.api.annotation.StartProcessTest;
+import org.camunda.bpm.engine.cdi.test.bpmn.SignalEventTest;
+import org.camunda.bpm.engine.cdi.test.bean.SignalEventTestBeans;
+import org.camunda.bpm.engine.cdi.test.impl.context.BusinessProcessContextTest;
+import org.camunda.bpm.engine.cdi.test.bean.BusinessProcessContextTestBeans;
+import org.camunda.bpm.engine.cdi.test.impl.context.MultiInstanceTest;
+import org.camunda.bpm.engine.cdi.test.bean.QuarkusLocalVariableBean;
+import org.camunda.bpm.engine.cdi.test.bean.QuarkusCdiTaskListenerBean;
+import org.camunda.bpm.engine.cdi.test.impl.el.TaskListenerInvocationTest;
+import org.camunda.bpm.engine.cdi.test.impl.event.EventNotificationTest;
+import org.camunda.bpm.engine.cdi.test.impl.event.TestEventListener;
+import org.jboss.arquillian.container.test.api.DeploymentConfiguration;
+import org.jboss.arquillian.container.test.api.DeploymentConfiguration.DeploymentContentBuilder;
+import org.jboss.arquillian.container.test.spi.client.deployment.AutomaticDeployment;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class QuarkusAutomaticDeployment implements AutomaticDeployment {
+
+  protected Map<Class<?>, Set<Class<?>>> beanDefinitions = new HashMap<Class<?>, Set<Class<?>>>() {{
+
+    Set<Class<?>> processControllerBean = asSet(QuarkusDeclarativeProcessController.class);
+
+    put(CompleteTaskTest.class, processControllerBean);
+    put(ProcessVariableLocalTypedTest.class, processControllerBean);
+    put(ProcessVariableTypedTest.class, processControllerBean);
+    put(StartProcessTest.class, processControllerBean);
+
+    Set<Class<?>> businessProcessContextBeans = asSet(BusinessProcessContextTestBeans.class);
+    put(BusinessProcessBeanTest.class, businessProcessContextBeans);
+    put(BusinessProcessContextTest.class, businessProcessContextBeans);
+
+    put(SignalEventTest.class, asSet(SignalEventTestBeans.class));
+    put(MultiInstanceTest.class, asSet(QuarkusLocalVariableBean.class));
+    put(TaskListenerInvocationTest.class, asSet(QuarkusCdiTaskListenerBean.class));
+    put(EventNotificationTest.class, asSet(TestEventListener.class));
```

Comment:
From this code, my understanding is that we will still need to declare some beans used by the `engine-cdi` test suite. Should we document this?

Reply:
Let's...
* Refactor the existing tests to always set explicitly the scope `Dependent.class`
* For the `@Unremoval` bean we add a code comment